test/tests/functional/pbs_preemption.py

```diff
@@ -251,3 +255,80 @@ def test_preempt_retry(self):
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
         self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+
+    def test_vnode_resource_contention(self):
+        """
+        Test to make sure that preemption happens when the resource in
+        contention is vnode.
+        """
+
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes(name='vnode', attrib=a, num=11,
```

Comment:
any particular reason why you are using 11 vnodes and 5 low priority jobs? Why not just 1 vnode and 1 low priority job?

Reply:
I wanted to have a sufficient amount of nodes with free ncpus so that scheduler does not apply preemption because of ncpus. 11 is just a random number, I could have done it with lesser. If you don't feel strongly about it, I would like to retain it as is.